numpy/_core/src/multiarray/iterators.c

```diff
@@ -695,6 +695,25 @@ iter_subscript(PyArrayIterObject *self, PyObject *ind)
         obj = ind;
     }
 
+    if (PyArrayIter_Check(obj)) {
+        PyArrayObject *tmp_arr = (PyArrayObject *)PyArray_FROM_O(obj);
+        if (tmp_arr == NULL) {
+            goto fail;
+        }
+
+        if (PyArray_SIZE(tmp_arr) == 0) {
```

Comment:
Would be good to explicitly test this branch. Should it run only if `obj` was not already an `ndarray`?